src/Badges.sol

```diff
@@ -102,6 +112,12 @@ contract Badges is
   }
 
   function createSpec(string memory _specUri, uint256 _raftTokenId) external virtual {
+    // replace the check that msg.sender is the owner of a RAFT
+    // with a check to see that msg.sender owns a badge associated with the given raftTokenId
+    require(
```

Comment:
Could we move this and the 2nd require into a new method called `private isAuthorizedToCreateSpec(_raftTokenId) returns bool` and maintain business logic there? Will it be compatible for upgrade?

Reply:
yep, changing it to a modifier makes sense. I don't see any reason why it would affect upgradeability